packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -1165,7 +1165,17 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
   /// Will throw an error if there is no back button in the page.
   Future<void> pageBack() async {
     return TestAsyncUtils.guard<void>(() async {
-      Finder backButton = find.byTooltip('Back');
+      Finder backButton = find.byType(BackButton);
+      if (backButton.evaluate().isEmpty) {
+        backButton = find.byElementPredicate((Element element) {
+          if (element.widget case final Tooltip tooltip) {
+            final MaterialLocalizations localizations = MaterialLocalizations.of(element);
+            return tooltip.message == localizations.backButtonTooltip;
```

Comment:
Any way to tie this also to the back icon? The tooltip is good, but also fairly generic (in English, it's just "Back", which could be a tooltip for some other control.

Reply:
I was trying to keep the previous behaviour for maximal backward compatibility.

But personally, I think adding `find.byType(BackButton)` is enough. The documentation only states "Makes an effort".

Maybe something like:
```dart
Finder backButton = find.byType(BackButton);
if (backButton.evaluate().isEmpty) {
  backButton = find.byType(CloseButton);
}
if (backButton.evaluate().isEmpty) {
  backButton = find.byType(CupertinoNavigationBarBackButton);
}
```

WDYT?

Reply:
Yes, that seems pretty good, although I wouldn't search for a close button, since that's a different thing (it doesn't always go "back").

Another fallback might be to search for the `BackButtonIcon`.  In fact, it might be better to search for that instead of `BackButton`, since `BackButton` and the action theme both use it. `CupertinoNavigationBarBackButton` should *probably* use something like it, but it doesn't.

It might be nice to factor out the finder here so the testing package provides a stock back button `Finder` in [`CommonFinders`](https://github.com/flutter/flutter/blob/7ffd36cb6dd8ed21949ac94157f73d8cb4c552e7/packages/flutter_test/lib/src/finders.dart#L59) that would find "the" back button when you use `find.backButton()`.  You could then use that here instead of putting the implementation here.

Reply:
Thanks, I followed your advise. Please take another look.